File: 2016/gara4/html/att/html.pas
function unescape(N: longint; var S: array of char): longint;
var i: longint;
begin
    (* Mettete qui il codice della soluzione *)
    for i:=0 to N - 1 do
        S[i] := 'x';

    unescape := 42;
end;


const
    MAXN = 1000010;
var
    N, i  : longint;
    S     : array[0..MAXN-1] of char;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);

    readln(fr, N);
    for i:=0 to N - 1 do
        read(fr, S[i]);

    N := unescape(N, S);
    writeln(fw, N);

    for i:=0 to N - 1 do
        write(fw, S[i]);

    close(fr);
    close(fw);
end.

File: ed2/gara4/html/att/html.c
#include <stdio.h>
#include <assert.h>

int unescape(int N, char S[]) {
    // Mettete qui il codice della soluzione
    int i;
    for (i=0; i<N; i++) {
        S[i] = 'x';
    }

    return 42;
}


#define MAXN 1000000

char S[MAXN];

int main() {
    FILE *fr, *fw;
    int N, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(1 == fscanf(fr, "%d\n", &N));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%c", &S[i]));

    N = unescape(N, S);
    fprintf(fw, "%d\n", N);
    for(i=0; i<N; i++)
        fprintf(fw, "%c", S[i]);
    fclose(fr);
    fclose(fw);
    return 0;
}
